cpp/recognition/cograph/BretscherCorneilHabibPaulCographRecognition.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+ * BretscherCorneilHabibPaulCographRecognition.cpp
+ *
+ *  Created on: 2024
+ *      Author: fixikmila
+ */
+// Copyright 2024 milana
+#include <graph/GraphTools.hpp>
+
+#include "recognition/CographRecognition.hpp"
+#include "recognition/CoTree.hpp"
+
+#include <list>
+
+namespace Koala {
+
+    bool BretscherCorneilHabibPaulCographRecognition::isCograph() const {
+        assureFinished();
+        return is_cograph;
+    }
+
+    void BretscherCorneilHabibPaulCographRecognition::run() {
+        hasRun = true;
+        std::vector<NetworKit::node> start;
+        for (auto u : graph.nodeRange()) {
+            start.push_back(u);
+        }
+        auto [bb, a] = LexBfsMinus(false, start);
+        auto [borders1, b] = LexBfsMinus(true, a);  // graph complement
+        auto [borders2, c] = LexBfsMinus(false, b);
+        if (NeighbourhoodSubsetProperty(true, b, borders1) &&
+        NeighbourhoodSubsetProperty(false, c, borders2)) {
+            is_cograph = true;
+        } else {
+            is_cograph = false;
+        }
+    }
+
+    std::pair<std::vector<std::vector<std::pair<int, int>>>, std::vector<NetworKit::node>>
```

Comment:
Well, the return type is horrible... my question is: maybe it's possible to define a local private named class with the same functionality (i.e. two fields: vec<vec<pair>> and vec<node>)? What do you think?

Reply:
fixed